src/main/kotlin/apply/application/ApplicantService.kt

```diff
@@ -16,22 +16,33 @@ import javax.annotation.PostConstruct
 @Transactional
 @Service
 class ApplicantService(
+    private val applicationFormService: ApplicationFormService,
     private val applicantRepository: ApplicantRepository,
     private val cheaterRepository: CheaterRepository,
     private val jwtTokenProvider: JwtTokenProvider
 ) {
-    fun findAll(): List<ApplicantResponse> = applicantRepository.findAll().map {
-        ApplicantResponse(it, cheaterRepository.existsByApplicantId(it.id))
+    fun findAll(): List<ApplicantBasicResponse> = applicantRepository.findAll().map {
+        ApplicantBasicResponse(it)
     }
 
-    fun findAllByIds(applicantIds: List<Long>): List<ApplicantResponse> =
-        applicantRepository.findAllById(applicantIds).map {
-            ApplicantResponse(it, cheaterRepository.existsByApplicantId(it.id))
+    fun findAllByRecruitmentId(recruitmentId: Long): List<ApplicantResponse> {
+        val applicationForms = applicationFormService.findAllByRecruitmentId(recruitmentId).map {
+            it.applicantId to it
+        }.toMap()
+
+        return applicantRepository.findAllById(applicationForms.keys).map {
+            ApplicantResponse(it, cheaterRepository.existsByApplicantId(it.id), applicationForms.getValue(it.id))
         }
+    }
+
+    fun findByRecruitmentIdAndKeyword(recruitmentId: Long, keyword: String): List<ApplicantResponse> {
+        return findAllByRecruitmentId(recruitmentId)
+            .filter { it.name.contains(keyword) || it.email.contains(keyword) }
```

Comment:
```suggestion
        return findByRecruitmentIdAndNameContainingOrEmailContaining(recruitmentId, keyword, keyword)
```
이런 방법도 가능합니다. 단순 제안이니 수정하지 않으셔도 돼요 :)
or이 and보다 우선순위가 높은 걸 활용하는 방법일 거에요!

Reply:
`Applicant`가 `recruitmentId`를 가지고 있지 않아서요.
먼저 `ApplicationForm`을 조회한 후에 거기서 포함되어 있는지를 확인하도록 로직을 짰습니다.

Reply:
아아... 죄송합니다